### PR TITLE
Add docs for minReadySeconds in StatefulSet

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
Add docs from minReadySeconds.

xref: https://github.com/kubernetes/enhancements/pull/2607

### CODE REVIEW

**Comment by kbhawkey** [comment]
Hi @ravisantoshgudimetla .
Is this change for the 1.22 release? Docs for v1.22 features should use the `dev-1.22` branch.
/cc @PI-Victor

**Comment by ravisantoshgudimetla** [comment]
> Hi @ravisantoshgudimetla .
> Is this change for the 1.22 release? Docs for v1.22 features should use the `dev-1.22` branch.
> /cc @PI-Victor

Yes @PI-Victor, updated the branch. Thank you.

**Review by soltysh — CHANGES_REQUESTED** [review]
Don't forget to update https://github.com/kubernetes/website/blob/master/content/en/docs/reference/command-line-tools-reference/feature-gates.md

**Review by soltysh — APPROVED** [review]
/lgtm

File: content/en/docs/concepts/workloads/controllers/statefulset.md [hunk 1]
```diff
@@ -228,6 +228,13 @@ and Ready or completely terminated prior to launching or terminating another
 Pod. This option only affects the behavior for scaling operations. Updates are not
 affected.
 
+#### Min Ready Seconds
```
Comment (tengqm) [inline content/en/docs/concepts/workloads/controllers/statefulset.md:231]: I'd suggest we spell this as "Minimum ready seconds".

Comment (ravisantoshgudimetla) [reply content/en/docs/concepts/workloads/controllers/statefulset.md]: Done. PTAL

**Comment by ravisantoshgudimetla** [comment]
@tengqm - Can you apply the required labels?

File: content/en/docs/concepts/workloads/controllers/statefulset.md [hunk 2]
```diff
@@ -228,6 +228,13 @@ and Ready or completely terminated prior to launching or terminating another
 Pod. This option only affects the behavior for scaling operations. Updates are not
 affected.
 
+#### Minimum Ready Seconds
+
+`.spec.minReadySeconds` is an optional field that specifies the minimum number of seconds for which a newly
```
Comment (tengqm) [inline content/en/docs/concepts/workloads/controllers/statefulset.md:233]: You may want to insert the following line before line 233:

```
{{< feature-state for_k8s_version="v1.22" state="alpha" >}}
```

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 3]
```diff
@@ -793,6 +794,8 @@ Each feature gate is designed for enabling/disabling a specific feature:
 - `StartupProbe`: Enable the
   [startup](/docs/concepts/workloads/pods/pod-lifecycle/#when-should-you-use-a-startup-probe)
   probe in the kubelet.
+- `StatefulSetMinReadySeconds`: Enable [minReadySeconds](/docs/concepts/workloads/controllers/statefulset.md#min-ready-seconds)
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:797]:
```suggestion
- `StatefulSetMinReadySeconds`: Enable [minReadySeconds](/docs/concepts/workloads/controllers/statefulset/#min-ready-seconds)
```

File: content/en/docs/concepts/workloads/controllers/statefulset.md (hunk 2 above)
Comment (ravisantoshgudimetla) [reply content/en/docs/concepts/workloads/controllers/statefulset.md:235]: Done. Please let me know if it is at right location.

**Comment by ravisantoshgudimetla** [comment]
I don't need to update the featuregates as it was already covered by @tengqm  in 372d34d

**Review by soltysh — APPROVED** [review]
/lgtm

**Review by sftim — COMMENTED** [review]
Do we also need to update https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/ to mention `StatefulSetMinReadySeconds`?

**Review by sftim — COMMENTED** [review]
Is this behind the `StatefulSetMinReadySeconds` feature gate? If so, let's also mention that in the StatefulSet concept page, _and_ update https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/

**Comment by chrisnegus** [comment]
> Is this behind the `StatefulSetMinReadySeconds` feature gate? If so, let's also mention that in the StatefulSet concept page, _and_ update https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/

I'm 1.22 release docs shadow for this PR. It looks like the changes for this feature to the [dev-1.22 feature-gates](https://github.com/kubernetes/website/blob/dev-1.22/content/en/docs/reference/command-line-tools-reference/feature-gates.md) page have already been merged. This PR seems to cover the StatefulSet concept page you asked about. @sftim is there more that you think needs to be done or is this ready to be merged?

**Comment by PI-Victor** [comment]
based on Chris' comment i'd like to merge this, anything to else needs to be added?

File: content/en/docs/concepts/workloads/controllers/statefulset.md [hunk 4]
```diff
@@ -228,6 +228,15 @@ and Ready or completely terminated prior to launching or terminating another
 Pod. This option only affects the behavior for scaling operations. Updates are not
 affected.
 
+#### Minimum Ready Seconds
+
+{{< feature-state for_k8s_version="v1.22" state="alpha" >}}
+
```
Comment (sftim) [inline content/en/docs/concepts/workloads/controllers/statefulset.md:234]: Let's explain that this field only works if you enable the `StatefulSetMinReadySeconds` feature gate.

**Review by sftim — CHANGES_REQUESTED** [review]
I think this feedback is right - not sure. If I'm wrong please comment and then dismiss this review.

File: content/en/docs/concepts/workloads/controllers/statefulset.md (hunk 4 above)
Comment (ravisantoshgudimetla) [reply content/en/docs/concepts/workloads/controllers/statefulset.md:234]: Sure, I added it [here](https://github.com/kubernetes/website/pull/27683/files#diff-ff635592a5e94bba6c6677379e9a5d64b4a035846dfe25881c1f05671e39b8bdR240). PTAL.

File: content/en/docs/concepts/workloads/controllers/statefulset.md [hunk 5]
```diff
@@ -228,6 +228,16 @@ and Ready or completely terminated prior to launching or terminating another
...
+`.spec.minReadySeconds` is an optional field that specifies the minimum number of seconds for which a newly
+created Pod should be ready without any of its containers crashing, for it to be considered available.
+This defaults to 0 (the Pod will be considered available as soon as it is ready). To learn more about when
+a Pod is considered ready, see [Container Probes](/docs/concepts/workloads/pods/pod-lifecycle/#container-probes).
+
+Please note that this field only works if you enable the `StatefulSetMinReadySeconds` featuregate
```
Comment (tengqm) [inline content/en/docs/concepts/workloads/controllers/statefulset.md:240]:
```suggestion
Please note that this field only works if you enable the `StatefulSetMinReadySeconds` feature gate.
```

Comment (ravisantoshgudimetla) [reply content/en/docs/concepts/workloads/controllers/statefulset.md]: Done. PTAL

**Comment by tengqm** [comment]
All feedbacks addressed. Good to go.
/lgtm
/approve

**Pull request merged**